File: app/src/main/java/com/nomdoa5/nomdo/adapter/BalanceAdapter.java
package com.nomdoa5.nomdo.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.nomdoa5.nomdo.R;
import com.nomdoa5.nomdo.model.BalanceModel;

import java.util.List;

public class BalanceAdapter extends RecyclerView.Adapter<BalanceAdapter.ViewHolder> {

    Context context;
    List<BalanceModel> balance_list;

    public BalanceAdapter(Context context, List<BalanceModel> balance_list){
        this.context = context;
        this.balance_list = balance_list;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_balance,parent , false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        if(balance_list != null && balance_list.size() > 0){
            BalanceModel model = balance_list.get(position);
            holder.tv_id.setText(model.getId());
            holder.tv_keterangan.setText(model.getKeterangan());
            holder.tv_income.setText(model.getIncome());
            holder.tv_outcome.setText(model.getOutcome());
        } else {
            return;
        }
    }

    @Override
    public int getItemCount() {
        return balance_list.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView tv_id, tv_keterangan, tv_income, tv_outcome;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            tv_id = itemView.findViewById(R.id.tv_id);
            tv_keterangan = itemView.findViewById(R.id.tv_keterangan);
            tv_income = itemView.findViewById(R.id.tv_income);
            tv_outcome = itemView.findViewById(R.id.tv_outcome);
        }
    }
}

File: app/src/main/java/com/nomdoa5/nomdo/helpers/adapter/BalanceAdapter.kt
package com.nomdoa5.nomdo.helpers.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.nomdoa5.nomdo.R
import com.nomdoa5.nomdo.helpers.toCurrencyFormat
import com.nomdoa5.nomdo.repository.model.Balance
import com.nomdoa5.nomdo.repository.model.Board

class BalanceAdapter(var context: Context,  private val listener: OnBalanceClickListener) :
    RecyclerView.Adapter<BalanceAdapter.ViewHolder>() {
    private val mData = ArrayList<Balance>()
    private var limit: Int? = null

    fun setData(items: ArrayList<Balance>) {
        mData.clear()
        mData.addAll(items)
        notifyDataSetChanged()
    }

    fun setLimit(count: Int){
        limit = count
    }

    interface OnBalanceClickListener {
        fun onBalanceClick(data: Balance)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_balance, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        if (mData.size > 0) {
            val model = mData[position]
            if (model.isIncome == 1) {
                holder.tvNominal.setTextColor(ContextCompat.getColor(context, R.color.primary))
            } else {
                holder.tvNominal.setTextColor(
                    ContextCompat.getColor(
                        context,
                        R.color.accent_profile
                    )
                )
            }

            holder.tvDescription.text = model.balanceDescription
            holder.tvNominal.text = model.nominal!!.toDouble().toCurrencyFormat()
            holder.itemView.setOnClickListener{listener.onBalanceClick(mData[position])}
        } else {
            return
        }
    }

    override fun getItemCount(): Int {
        return if(limit != null){
            limit!!
        } else {
            mData.size
        }
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvDescription: TextView = itemView.findViewById(R.id.tv_description_balance)
        var tvDate: TextView = itemView.findViewById(R.id.tv_date_balance)
        var tvNominal: TextView = itemView.findViewById(R.id.tv_nominal_balance)
    }
}